src/common/libzmqutil/test/cert.c

```diff
@@ -0,0 +1,308 @@
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <stdbool.h>
+#include <errno.h>
+
+#include "tap.h"
+#include "ccan/str/str.h"
+#include "ccan/array_size/array_size.h"
+
+#include "cert.h"
+
+struct test_vec {
+    const char *name;
+    char *input;
+};
+
+static struct test_vec goodvec[] = {
+    {
+        .name = "czmq zcert sample",
+        .input =
+    "#   ****  Generated on 2023-09-16 23:15:27 by CZMQ  ****\n"
+    "#   ZeroMQ CURVE **Secret** Certificate\n"
+    "#   DO NOT PROVIDE THIS FILE TO OTHER USERS nor change its permissions.\n"
+    "\n"
+    "metadata\n"
+    "    name = \"picl0\"\n"
+    "    keygen.czmq-version = \"4.2.1\"\n"
+    "    keygen.sodium-version = \"1.0.18\"\n"
+    "    keygen.flux-core-version = \"0.54.0\"\n"
+    "    keygen.hostname = \"picl0\"\n"
+    "    keygen.time = \"2023-09-16T23:15:27\"\n"
+    "    keygen.userid = \"5588\"\n"
+    "    keygen.zmq-version = \"4.3.4\"\n"
+    "curve\n"
+    "    public-key = \"S:do@!Xbon>XQ$e!SKvqor&X>apyf[ZtG@(v^9*e\"\n"
+    "    secret-key = \"pwS@!apyf=&@1Kvorq7uv]sb{*x!v.alB.199AD6\"\n"
+    },
+    {
+         .name = "cert with blank lines",
+         .input =
+    "curve\n"
+    "\n"
+    " public-key = \"S:do@!Xbon>XQ$e!SKvqor&X>apyf[ZtG@(v^9*e\"\n"
+    "\n"
+    " secret-key = \"pwS@!apyf=&@1Kvorq7uv]sb{*x!v.alB.199AD6\"\n"
+    },
+    {
+         .name = "cert with indented inline comments",
+         .input =
+    "metadata\n"
+    "	# comment \n"
+    "curve\n"
+    "# comment \n"
+    " public-key = \"FYFE.@650VuUqRGygAtG.RC$A<cid63q(WEnR+&y\"\n"
+    " secret-key = \"^Oto<5V$^d/V}kj6^Q[LRwdgAY5K3gE{gN2+1ox)\"\n"
+    },
+    {
+         .name = "cert with # in z85 key",
+         .input =
+    "curve\n"
+    " public-key = \"s70JW1{s!)bET!S&yF=7z=b{%+<2Nu1zO31tCadW\"\n"
+    " secret-key = \"*B!T$M-&ULFq)5*@Wo+/o*#7P/DVcr:Xz2a5PyMG\"\n"
+    },
+};
+
+static struct test_vec badvec[] = {
+    {
+        .name = "empty input",
+        .input = ""
+    },
+    {
+        .name = "cert with missing curve section",
+        .input =
+    "metadata\n"
+    },
+    {
+        .name = "cert with empty curve section",
+        .input =
+    "metadata\n"
+    "curve\n"
+    },
+    {
+        .name = "cert with extra section",
+        .input =
+    "metadata\n"
+    "unknown\n"
+    "curve\n"
+    "  public-key = \".0TmVO:u.]cUa[tFi[rH<3q2Cmu?1IBq!8/Q+&(0\"\n"
+    "  secret-key = \"UPk352x.N&a(0mXR&{9S^<I!cH[Oo*^BeU)UD/[5\"\n"
+    },
+    {
+        .name = "cert with curve section indented",
+        .input =
+    " curve\n"
+    " public-key = \".0TmVO:u.]cUa[tFi[rH<3q2Cmu?1IBq!8/Q+&(0\"\n"
+    " secret-key = \"UPk352x.N&a(0mXR&{9S^<I!cH[Oo*^BeU)UD/[5\"\n"
+    },
+    {
+        .name = "cert with keys not indented",
+        .input =
+    "curve\n"
+    "public-key = \".0TmVO:u.]cUa[tFi[rH<3q2Cmu?1IBq!8/Q+&(0\"\n"
+    "secret-key = \"UPk352x.N&a(0mXR&{9S^<I!cH[Oo*^BeU)UD/[5\"\n"
+    },
+};
```

Comment:
Might be good to expand the "bad" tests here, including something like a "good" cert with one character change (so should fail), some garbage input, public and/or secret key too long, unclosed quote, etc